Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -4160,4 +4165,48 @@ public int hashCode() {
         hash = 89 * hash + Objects.hashCode(this.playerId);
         return hash;
     }
+
+
+    @Override
+    public void addPhyrexianToColors(FilterMana colors) {
+        if (colors.isWhite()) {
+            this.phyrexianColors.setWhite(true);
+        }
+        if (colors.isBlue()) {
+            this.phyrexianColors.setBlue(true);
+        }
+        if (colors.isBlack()) {
+            this.phyrexianColors.setBlack(true);
+        }
+        if (colors.isRed()) {
+            this.phyrexianColors.setRed(true);
+        }
+        if (colors.isGreen()) {
+            this.phyrexianColors.setGreen(true);
+        }
+    }
+
+    @Override
+    public void removePhyrexianFromColors(FilterMana colors) {
+        if (colors.isWhite()) {
+            this.phyrexianColors.setWhite(false);
+        }
+        if (colors.isBlue()) {
+            this.phyrexianColors.setBlue(false);
+        }
+        if (colors.isBlack()) {
+            this.phyrexianColors.setBlack(false);
+        }
+        if (colors.isRed()) {
+            this.phyrexianColors.setRed(false);
+        }
+        if (colors.isGreen()) {
+            this.phyrexianColors.setGreen(false);
+        }
+    }
+    
+    @Override
+    public FilterMana getPhyrexianColors() {
+        return this.phyrexianColors.copy();
```

Comment:
Why it's makes copies in every getter call instead simple ref? Normal code: return ref objects and create copy on Copy/Create method (but you are using ref in copy -- see above).